### PR TITLE
Add more text to top of Client API

### PR DESCRIPTION
**Pull request by martindurant**
Extends #6793

@jsignell : I meant something more verbose like this.

### CODE REVIEW

**Review by jsignell — COMMENTED** [review]
Ah ok if it's this much verbosity that you are looking for then I think it belongs in the docstring itself or in some other place in the documentation: probably client.rst. Otherwise this feels like an overloading of the API.rst.

**Comment by martindurant** [comment]
I think it needs to be here, because this is probably the single most accessed page in the distributed documentation - we should have stats on that.

File: docs/source/api.rst [hunk 1]
```diff
@@ -7,6 +7,24 @@ API
...
+cluster it is connected to. Simply creating a client may be all that a users needs to do,
+since it will implicitly become the default engine for any dask ``compute()`` calls.
+
+There are three main ways in which ``Client`` can be called:
+- with the address of an existing scheduler, or scheduler information file; in this case,
+  the cluster has been configured elsewhere (e.g., command line, helm)
+- with a cluster instance which has been created in the current python session; in this case,
+  you have control over the cluster object (which may point to Kubernetes, for example), and
+  you may call methods on it to, for instance, scale.
+- without any connection argument; in this case, a :class:`distributed.LocalCluster` will be
```
Comment (jacobtomlinson) [inline docs/source/api.rst:21]: It will also check the `DASK_SCHEDULER_ADDRESS` env var and if it is set it will connect to that address.

Comment (martindurant) [reply docs/source/api.rst:21]: I didn't even know that.

**Pull request closed without merging**